application/controllers/OptoutController.php

```
@@ -175,8 +258,8 @@ public function actionremovetokens()
         if (!isset($token)) {
             $message = gT('You are not a participant in this survey.');
         } else {
-            if (substr((string) $token->emailstatus, 0, strlen('OptOut')) !== 'OptOut') {
-                $token->emailstatus = 'OptOut';
+            if (!$token->optOutStatus) {
+                $oToken->optOut();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$oToken'. (undefined)

(UnusedLocalVariable)